src/flamegraph/attrs.rs

```diff
@@ -60,13 +62,16 @@ impl FuncFrameAttrsMap {
                     value = &value[1..value.len() - 1];
                 }
                 match name {
-                    "title" => funcattrs.title = Some(value.to_string()),
-                    "id" => funcattrs.g.id = Some(value.to_string()),
-                    "class" => funcattrs.g.class = Some(value.to_string()),
-                    "href" => funcattrs.a.href = Some(value.to_string()),
-                    "target" => funcattrs.a.target = Some(value.to_string()),
-                    "g_extra" => parse_extra_attrs(&mut funcattrs.g.extra, value),
-                    "a_extra" => parse_extra_attrs(&mut funcattrs.a.extra, value),
+                    "title" => {
+                        funcattrs.title = Some(value.to_string());
+                    }
+                    "href" => {
+                        funcattrs.add_attr(func, "xlink:href".to_string(), value.to_string());
+                    }
+                    "id" | "class" | "target" => {
+                        funcattrs.add_attr(func, name.to_string(), value.to_string());
```

Comment:
Double the number of `String`s... I guess it doesn't really matter since this isn't in performance-sensitive code.

Reply:
Yes, for the non-extra attributes it will be double the number of `String`s and I don't think it matters at all for loading nameattr files. It makes everything else simpler to put all of the attributes in a map.